Library/Homebrew/extend/os/mac/hardware.rb

```diff
@@ -24,4 +24,20 @@ def self.oldest_cpu(version = nil)
       generic_oldest_cpu
     end
   end
+
+  # Override
+  # Mirrors version-dependent logic of oldest_cpu
+  sig { params(version: T.nilable(Version)).returns(String) }
+  def self.rustflags_target_cpu(version = nil)
```

Comment:
When would `version` be passed through here rather than just use the current `MacOS.version`?